src/mlpack/core/util/param_checks_impl.hpp

```diff
@@ -306,6 +306,29 @@ inline void ReportIgnoredParam(util::Params& params,
   }
 }
 
+// Check if the given input data points aren't empty.
+inline void RequireNonEmptyInputValue(
+    util::Params& params,
+    const std::string& paramName,
+    const bool fatal,
+    const std::string& errorMessage)
+{
+  if (BINDING_IGNORE_CHECK(paramName))
+    return;
+
+  if (params.Has(paramName))
+  {
+    arma::mat trainingData = std::move(params.Get<arma::mat>(paramName));
+    if (trainingData.empty())
+    {
+      util::PrefixedOutStream& stream = fatal ? Log::Fatal : Log::Warn;
+      if (!errorMessage.empty())
+        stream << paramName << " " << errorMessage << "!" << std::endl;
```

Comment:
You should use `PRINT_PARAM_STRING()` or similar here.  Probably it would also be a good idea to print the value of the parameter (so e.g. if the language is the command-line bindings, then the user will see the filename they specified).

Reply:
yes, that's better.